test/integ/routers/alpha-router/alpha-router.integration.test.ts

```diff
@@ -0,0 +1,995 @@
+          const tokenOut = Ether.onChain(1) as Currency;
+          const amount = tradeType == TradeType.EXACT_INPUT ?
+            parseAmount('1000000', tokenIn)
+            : parseAmount('10', tokenOut);
+
+          const swap = await alphaRouter.route(
+            amount,
+            getQuoteToken(tokenIn, tokenOut, tradeType),
+            tradeType,
+            {
+              recipient: alice._address,
+              slippageTolerance: SLIPPAGE,
+              deadline: parseDeadline(360),
+            },
+            {
+              ...ROUTING_CONFIG
+            }
+          );
+          expect(swap).toBeDefined();
+          expect(swap).not.toBeNull();
+
+          const {
+            quoteDecimals,
+            quoteGasAdjustedDecimals,
+            methodParameters
+          } = convertSwapDataToResponse(amount, tradeType, swap!)
+
+          await validateStandardSwapRoute(quoteDecimals, quoteGasAdjustedDecimals, tradeType, false);
+          expect(methodParameters).not.toBeUndefined;
+        })
+
+        it(`erc20 -> eth large trade`, async () => {
+          // Trade of this size almost always results in splits.
+          const tokenIn = USDC_MAINNET;
+          const tokenOut = Ether.onChain(1) as Currency;
+          const amount = tradeType == TradeType.EXACT_INPUT ?
+            parseAmount('1000000', tokenIn)
+            : parseAmount('100', tokenOut);
+
+          const swap = await alphaRouter.route(
+            amount,
+            getQuoteToken(tokenIn, tokenOut, tradeType),
+            tradeType,
+            {
+              recipient: alice._address,
+              slippageTolerance: SLIPPAGE,
+              deadline: parseDeadline(360),
+            },
+            {
+              ...ROUTING_CONFIG
+            }
+          );
+          expect(swap).toBeDefined();
+          expect(swap).not.toBeNull();
+
+          const {
+            quote,
+            methodParameters,
+            route,
+          } = convertSwapDataToResponse(amount, tradeType, swap!)
+
+          expect(methodParameters).not.toBeUndefined;
+          expect(route).not.toBeUndefined
+
+          const amountInEdgesTotal = _(route)
+            .flatMap((route) => route[0]!)
+            .filter((pool) => !!pool.amountIn)
+            .map((pool) => BigNumber.from(pool.amountIn))
+            .reduce((cur, total) => total.add(cur), BigNumber.from(0))
+          const amountIn = BigNumber.from(quote)
+          expect(amountIn.eq(amountInEdgesTotal))
+
+          const amountOutEdgesTotal = _(route)
+            .flatMap((route) => route[0]!)
+            .filter((pool) => !!pool.amountOut)
+            .map((pool) => BigNumber.from(pool.amountOut))
+            .reduce((cur, total) => total.add(cur), BigNumber.from(0))
+          const amountOut = BigNumber.from(quote)
+          expect(amountOut.eq(amountOutEdgesTotal))
+
+          const { tokenInBefore, tokenInAfter, tokenOutBefore, tokenOutAfter } = await executeSwap(
```

Comment:
could we do `validateStandardExecuteSwap` ? can just make it more customizeable? ie. take in the amounts to validate instead of hardcoding

Reply:
good idea, adding two optional params into the func for checking against amountIN and against amountOut. Also renaming to `validateExecuteSwap`